scylla-cdc/src/stream_generations.rs

```diff
@@ -0,0 +1,531 @@
+        let mut query = Query::new(format!(
+            "CREATE KEYSPACE IF NOT EXISTS {} WITH replication
+                = {{'class':'SimpleStrategy', 'replication_factor': 3}};",
+            TEST_KEYSPACE
+        ));
+        query.set_consistency(Consistency::All);
+
+        session.query(query, &[]).await.unwrap();
+
+        // Create test tables containing information about generations and streams.
+        for query in vec![
+            construct_generation_table_query(),
+            construct_stream_table_query(),
+        ] {
+            session.query(query, &[]).await.unwrap();
+        }
+
+        // Delete all leftovers from previous tests.
+        for table in vec![
+            TEST_STREAM_TABLE.to_string(),
+            TEST_GENERATION_TABLE.to_string(),
+        ] {
+            session
+                .query(format!("TRUNCATE {};", table), &[])
+                .await
+                .unwrap();
+        }
+    }
+
+    // Populate test tables with given data.
+    async fn populate_test_db(session: &Session) {
+        for generation in &[GENERATION_NEW_MILLISECONDS, GENERATION_OLD_MILLISECONDS] {
+            session
+                .query(
+                    format!(
+                        "INSERT INTO {} (key, time, expired) VALUES ('timestamps', ?, NULL);",
+                        TEST_GENERATION_TABLE
+                    ),
+                    (Timestamp(chrono::Duration::milliseconds(*generation)),),
+                )
+                .await
+                .unwrap();
+
+            session
+                .query(
+                    format!(
+                        "INSERT INTO {}(time, range_end, streams) VALUES (?, -1, {{{}, {}}});",
+                        TEST_STREAM_TABLE, TEST_STREAM_1, TEST_STREAM_2
+                    ),
+                    (Timestamp(chrono::Duration::milliseconds(
+                        GENERATION_NEW_MILLISECONDS,
+                    )),),
+                )
+                .await
+                .unwrap();
+        }
+    }
+
+    // Create setup for tests.
+    async fn setup() -> Result<GenerationFetcher, Box<dyn Error>> {
+        let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+
+        let session = SessionBuilder::new().known_node(uri).build().await?;
+
+        create_test_db(&session).await;
+        populate_test_db(&session).await;
+
+        let generation_fetcher = GenerationFetcher::test_new(&Arc::new(session));
+
+        Ok(generation_fetcher)
+    }
+
+    #[tokio::test]
+    async fn test_fetch_all_generations() {
+        let fetcher = setup().await.unwrap();
+
+        let correct_gen = vec![
+            Generation(chrono::Duration::milliseconds(GENERATION_NEW_MILLISECONDS)),
+            Generation(chrono::Duration::milliseconds(GENERATION_OLD_MILLISECONDS)),
+        ];
+
+        let gen = fetcher.fetch_all_generations().await.unwrap();
+
+        assert_eq!(gen.len(), correct_gen.len());
```

Comment:
Nit: no need to compare lengths of `gen` and `correct_gen` if you are comparing the whole values later (`assert_eq!(gen, correct_gen)`).